### DISCUSSION TITLE
v20.1.0

### DISCUSSION THREAD

**Discussion by fballiano** [OpenMage/magento-lts#3422] [Releases]
## You should absolutely know
Since the approval of [our second RFC - release schedule](https://github.com/OpenMage/rfcs/blob/main/accepted/0002-release-schedule.md)  OpenMage 19.x enters and "patch only" state, it will be maintained for two more years as promised but only _significant security patches or regression fixes_ will be ported to v19, every other development (and we have many) will be focused on v20+.

We encourage everybody to upgrade to v20, it is our latest and greatest and deserves the bit of work necessary for the upgrade (ask your developer/agency, don't do it yourself).

## Release highlights

This is a big release, that's why we decided to move away from the 20.0.x versioning and go to 20.1.x. Since a lot of changes could have some impact on current installations we decided to release some "rc" versions before the official 20.1.0. Tests are more than welcome now but be extra careful with production environment.

What's most important is the removal of all the 3rd party libraries (phpseclib, mcrypt_compat, Cm_RedisSession, Cm_Cache_Backend_Redis and Pelago_Emogrifier and Zend Framework) form our repository, they are now imported via composer. This was an important step to clean up and modernise our code.
* If you want to clean up your project from all the files that were removed during this release cycle, please [check this link](https://github.com/OpenMage/magento-lts/discussions/3170#discussioncomment-5722151) for the complete list.

Don't worry though, if you've always installed OpenMage extracting the zip file, starting from this release you'll find a new zip file attached to the release itself, we build this zip adding all of the old 3rd party libraries so that you will not have to migrate to composer or use composer at all.

Also:
- PHP 7.4 is now the minimum required version and 8.2 is now supported.
- The M1 legacy themes have been moved to an external repository since it's old (and mostly unused) code.
- A great improvement to EAV config cache has been added to v20.
- Support for Google Analytics 4 and Google Tag Manager was added with enhanced ecommerce data (which historically was always provided by 3rd party extensions).
- Possibility to set backend locale per every admin user was added.
- "pub/" folder structure was introduced (optional), to have extra security for your installation
- Massive performance improvement in complex SQL queries (#2210)
- Native WebP support
- Many, many smaller bugfix and improvements

Things you should be aware of:
- Support for Google Analytics3 and Universal Analytics were removed.
- Support for UPS CGI API (which was shut down in May 2023) was removed.
- Old inactive quotes are now actually purged from the database (#1489).

Known issues:
- The "pub/" folder structure won't work in composer managed projects, because the composer-magento plugin can't manage symlinks inside "pub/" correctly (#1210).

## Changelog
* Fix methods signature, ref #1610 by @fballiano in https://github.com/OpenMage/magento-lts/pull/2838
* Merge v19 -> v20 by @fballiano in https://github.com/OpenMage/magento-lts/pull/2766
* New event so more validation classes can be added on the fly by @woutersamaey in https://github.com/OpenMage/magento-lts/pull/1490
* Revert "New event so more validation classes can be added on the fly" by @fballiano in https://github.com/OpenMage/magento-lts/pull/2910
* Remove legacy media uploader / editor remnants by @justinbeaty in https://github.com/OpenMage/magento-lts/pull/2434
* Remove more Internet Explorer code by @justinbeaty in https://github.com/OpenMage/magento-lts/pull/2427
* EAV Config Cache by @davidhiendl in https://github.com/OpenMage/magento-lts/pull/2993
* EAV Config Cache followup: select non-existent attribute exception by @davidhiendl in https://github.com/OpenMage/magento-lts/pull/3038
* fix: cast store config (checkout/cart/delete_quote_after) to int to avoid unsupported operand types php error by @darinda in https://github.com/OpenMage/magento-lts/pull/3065
* Eav config cache followup for multi scope load by @davidhiendl in https://github.com/OpenMage/magento-lts/pull/3044
* EAV Config Cache revert return empty attribute model by @davidhiendl in https://github.com/OpenMage/magento-lts/pull/3074
* Removed module Mage_Poll by @fballiano in https://github.com/OpenMage/magento-lts/pull/3098
* Upgraded ExtJS to 1.1.1 by @justinbeaty in https://github.com/OpenMage/magento-lts/pull/2473
* Bump openmage/dev-meta-package from 1.0.2 to 1.0.3 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3138
* Updated branches for codeql analysis workflow by @fballiano in https://github.com/OpenMage/magento-lts/pull/3141
* Regenerated PHPStan baseline for "main" branch by @fballiano in https://github.com/OpenMage/magento-lts/pull/3142
* Fixed some minor indentation problem for docblocks by @fballiano in https://github.com/OpenMage/magento-lts/pull/3143
* Ended the request (and close the session) before core_app_run_after event by @colinmollenhour in https://github.com/OpenMage/magento-lts/pull/1592
* Added possibility to set backend locale per every admin user by @fballiano in https://github.com/OpenMage/magento-lts/pull/3087
* Used session_status() instead of $_SESSION in Mage_Core_Model_App by @fballiano in https://github.com/OpenMage/magento-lts/pull/3145
* Added check for coupon expiration date by @fballiano in https://github.com/OpenMage/magento-lts/pull/3144
* Improved global search performance by using only prefix matching. by @colinmollenhour in https://github.com/OpenMage/magento-lts/pull/1596
* Avoided calling getBackendModelByFieldConfig with empty field by @fballiano in https://github.com/OpenMage/magento-lts/pull/3151
* Fixed typo in the exception message by @ADDISON74 in https://github.com/OpenMage/magento-lts/pull/3153
* Avoided unnecessary calls to string translation methods for catalog/sales rules labels by @luigifab in https://github.com/OpenMage/magento-lts/pull/3140
* Updated README.md to reflect new RFC 0002 - Release Schedule by @colinmollenhour in https://github.com/OpenMage/magento-lts/pull/3139
* Bump phpstan/phpstan from 1.10.10 to 1.10.11 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3156
* Reduced getWebsiteId() calls by @luigifab in https://github.com/OpenMage/magento-lts/pull/3154
* Move login page title to controller by @luigifab in https://github.com/OpenMage/magento-lts/pull/3157
* Updated ACL of order objects and updated setActiveMenu for sales and catalog items by @luigifab in https://github.com/OpenMage/magento-lts/pull/3159
* [PHP 8.2] Fixed some deprecation warnings by @fballiano in https://github.com/OpenMage/magento-lts/pull/3155
* Reduced calls to char() method by @luigifab in https://github.com/OpenMage/magento-lts/pull/3161
* Fixed some whitespace issues in docblocks by @fballiano in https://github.com/OpenMage/magento-lts/pull/3162
* Added order tracking for Google Analytics 4 by @fballiano in https://github.com/OpenMage/magento-lts/pull/3092
* Fixed PHP warnings when session backend exits before returning. by @colinmollenhour in https://github.com/OpenMage/magento-lts/pull/3109
* Fixed price filter when search by non-numeric value by @sreichel in https://github.com/OpenMage/magento-lts/pull/3136
* Use default configurable attribute label for new products by @mmenozzi in https://github.com/OpenMage/magento-lts/pull/3168
* Fixed session renew timestamp should be updated when customer changes password by @colinmollenhour in https://github.com/OpenMage/magento-lts/pull/2916
* Removed all @author tags from docblocks by @fballiano in https://github.com/OpenMage/magento-lts/pull/3171
* Removed all @author tags from phpdocs (updated copyright docblocks) by @fballiano in https://github.com/OpenMage/magento-lts/pull/3172
* [PHP 8.2] Fixed deprecation warning in Mage_CatalogInventory_Model_Observer by @fballiano in https://github.com/OpenMage/magento-lts/pull/3178
* Ignore addresses when country is not authorized by @luigifab in https://github.com/OpenMage/magento-lts/pull/3019
* Bump phpstan/phpstan from 1.10.11 to 1.10.13 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3186
* Bump phpunit/phpunit from 9.6.6 to 9.6.7 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3185
* Fixed ESI parsing error with turpentine by @luigifab in https://github.com/OpenMage/magento-lts/pull/3183
* Fixed "object could not be converted to int" in Mage_Adminhtml_Block_Catalog_Product_Edit_Tab_Super_Config_Simple by @luigifab in https://github.com/OpenMage/magento-lts/pull/3182
* Fixed deprecation warning in Mage_Downloadable_Helper_Catalog_Product_Configuration by @fballiano in https://github.com/OpenMage/magento-lts/pull/3187
* Fixed deprecation warning in Mage_SalesRule_Model_Rule by @fballiano in https://github.com/OpenMage/magento-lts/pull/3188
* Set sort order and disable click-to-top for coupons grid by @luigifab in https://github.com/OpenMage/magento-lts/pull/3189
* [PHP 8.2] Fixed deprecation error for creation of dynamic property by @fballiano in https://github.com/OpenMage/magento-lts/pull/3177
* Update review_entity_summary when unapprove the latest comment by @luigifab in https://github.com/OpenMage/magento-lts/pull/2686
* Reduced number of getId() calls in Mage_Eav_Model_Entity_Type by @luigifab in https://github.com/OpenMage/magento-lts/pull/3194
* Fix for export failing when product_id missing from cataloginventory_stock_item by @discountscott in https://github.com/OpenMage/magento-lts/pull/3165
* Replaced Mage_Exception usages with Mage_Core_Exception by @fballiano in https://github.com/OpenMage/magento-lts/pull/3206
* Added patch for Zend_Mail and "emails are displayed incorrectly" problem by @fballiano in https://github.com/OpenMage/magento-lts/pull/3202
* Bump phpstan/phpstan from 1.10.13 to 1.10.14 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3211
* Updated docker-compose to mysql8 and redis7 by @fballiano in https://github.com/OpenMage/magento-lts/pull/3209
* Avoid JS errors if the zipcode field has no label by @azambon in https://github.com/OpenMage/magento-lts/pull/3197
* Fix TypeError: Unsupported operand types: string - string in Zend/Locale/Math/PhpMath.php by @colinmollenhour in https://github.com/OpenMage/magento-lts/pull/3217
* Removed PHPStan stubs by @fballiano in https://github.com/OpenMage/magento-lts/pull/3210
* Converted msrp_rwd.js to vanilla javascript by @fballiano in https://github.com/OpenMage/magento-lts/pull/3214
* Composer autoloader patch was added to app/Mage.php by @fballiano in https://github.com/OpenMage/magento-lts/pull/3216
* Fixed deprecation warning on Mage_Catalog_Helper_Image by @fballiano in https://github.com/OpenMage/magento-lts/pull/3201
* Fixed logging too many exceptions for payment methods by @fballiano in https://github.com/OpenMage/magento-lts/pull/3181
* Fixed autoloader patch to work with composer-installed setup by @colinmollenhour in https://github.com/OpenMage/magento-lts/pull/3226
* Converted msrp_rwd.js to vanilla javascript (part 2) by @fballiano in https://github.com/OpenMage/magento-lts/pull/3230
* Added type hint docblock for Google Analytics 4 template by @fballiano in https://github.com/OpenMage/magento-lts/pull/3232
* Added cache to "order status" labels in Mage_Sales_Model_Order_Config by @elidrissidev in https://github.com/OpenMage/magento-lts/pull/3223
* Fixed price expression SQL in layered navigation by @elidrissidev in https://github.com/OpenMage/magento-lts/pull/3227
* Fixed stock status update from SOAP API V2 by @elidrissidev in https://github.com/OpenMage/magento-lts/pull/1635
* Load all child blocks in Adminhtml Head block by @elidrissidev in https://github.com/OpenMage/magento-lts/pull/3222
* Moved isCurrentlySecure() method to Mage_Core_Model_App by @colinmollenhour in https://github.com/OpenMage/magento-lts/pull/3225
* Added cron to remove old index events for manual indexers by @tobihille in https://github.com/OpenMage/magento-lts/pull/2474
* Fixed model hasDataCchanges is true right after loading by @fballiano in https://github.com/OpenMage/magento-lts/pull/3243
* Removed reference to unexisting class Varien_Image_Adapter_Imagemagic_External by @fballiano in https://github.com/OpenMage/magento-lts/pull/3242
* Ensure store-scoped values are selected before default by @mattdavenport in https://github.com/OpenMage/magento-lts/pull/2966
* Added configuration for captcha failed_attempts_ip in backend by @damien-biasotto in https://github.com/OpenMage/magento-lts/pull/1074
* Removed "MAGE" cache tag by @fballiano in https://github.com/OpenMage/magento-lts/pull/3246
* Improved handling of ignore_user_agents by @empiricompany in https://github.com/OpenMage/magento-lts/pull/3238
* Fixed "if newsletter module disabled, customer backend would blank without notice" by @sidealice in https://github.com/OpenMage/magento-lts/pull/3078
* Also change the default when calling setTheme by @Asfolny in https://github.com/OpenMage/magento-lts/pull/1736
* Allowed "force enabling" of flat tables use for a product collection by @loekvangool in https://github.com/OpenMage/magento-lts/pull/2954
* Bump phpunit/phpunit from 9.6.7 to 9.6.8 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3261
* Bump phpstan/phpstan from 1.10.14 to 1.10.15 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3260
* Fixed PHPStan error in Mage_HTTP_Client_Socket and Mage_HTTP_Client_Curl by @fballiano in https://github.com/OpenMage/magento-lts/pull/3205
* Fixed undefined constant in installer by @sreichel in https://github.com/OpenMage/magento-lts/pull/3259
* Fixed Undefined variable $res in Mage_Core_Model_Cache by @fballiano in https://github.com/OpenMage/magento-lts/pull/3255
* Minor phpstan-fixes for php8.1 by @sreichel in https://github.com/OpenMage/magento-lts/pull/3237
* Fixed the behaviour of $forceLog parameter for Mage::log() by @empiricompany in https://github.com/OpenMage/magento-lts/pull/3263
* Added support for Google Analytics 4 Enhanced eCommerce by @discountscott in https://github.com/OpenMage/magento-lts/pull/3198
* Optimized count queries by removing all joins if there are no where by @colinmollenhour in https://github.com/OpenMage/magento-lts/pull/2210
* Added PHP8 configurations to .htaccess by @fballiano in https://github.com/OpenMage/magento-lts/pull/3254
* GitHub Actions - `check-files` workflow - split into steps as a workaround for step template size limit by @gorbunovav in https://github.com/OpenMage/magento-lts/pull/3276
* Fix admin ACL mismatch for OAuth (#3272) by @Tomasz-Silpion in https://github.com/OpenMage/magento-lts/pull/3274
* Fixed PHP8.2 deprecated warning in Mage_Cron_Model_Observer by @fballiano in https://github.com/OpenMage/magento-lts/pull/3253
* Introduced pub/ structure for webserver home folder mapping by @speedupmate in https://github.com/OpenMage/magento-lts/pull/1210
* Bump tj-actions/changed-files from 35 to 36 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3286
* Bump friendsofphp/php-cs-fixer from 3.16.0 to 3.17.0 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3285
* Fixed deprecation warning in Mage_Core_Model_Url::isOwnOriginUrl() by @fballiano in https://github.com/OpenMage/magento-lts/pull/3284
* Escape product names and translations inside js by @Judx in https://github.com/OpenMage/magento-lts/pull/2673
* Removed auto_detect_line_endings since it's deprecated in 8.1 by @fballiano in https://github.com/OpenMage/magento-lts/pull/3282
* Replaced strpos() calls with PHP8 string functions by @sreichel in https://github.com/OpenMage/magento-lts/pull/3258
* Fix regression from getSelectCountSql optimization when collection us… by @colinmollenhour in https://github.com/OpenMage/magento-lts/pull/3279
* Mage_Rules now truncates strings after 100 characters instead of 33 in the backend by @loekvangool in https://github.com/OpenMage/magento-lts/pull/3288
* Added back Order add comment functionality to REST API by @elidrissidev in https://github.com/OpenMage/magento-lts/pull/3292
* Fixed E_DEPRECATED str_replace(): Passing null in Translate.php by @kiatng in https://github.com/OpenMage/magento-lts/pull/3294
* Fixed passing null is deprecated for trim in Mage_Tax_Model_Resource_Calculation by @kyrena in https://github.com/OpenMage/magento-lts/pull/3299
* Fixed documentation on how to contribute after branch change by @fballiano in https://github.com/OpenMage/magento-lts/pull/3306
* Fixed passing null is deprecated for stripos in Mage_Core_Model_Resource_Helper_Abstract by @kyrena in https://github.com/OpenMage/magento-lts/pull/3295
* Fixed passing null is deprecated for explode in Mage_Catalog_Model_Resource_Product_Indexer_Eav_Source by @kyrena in https://github.com/OpenMage/magento-lts/pull/3297
* Allowed system config of type "serialized array" to <depend> on another field by @pquerner in https://github.com/OpenMage/magento-lts/pull/3307
* Fixed passing null is deprecated for htmlspecialchars in Mage_Adminhtml_Block_System_Config_Form_Field by @kyrena in https://github.com/OpenMage/magento-lts/pull/3296
* "Product Tags" tab is hidden in customer admin mask if Mage_Tag is disabled by @kyrena in https://github.com/OpenMage/magento-lts/pull/3298
* Fix #3312 ExtJS Tree Checkbox Update by @justinbeaty in https://github.com/OpenMage/magento-lts/pull/3314
* Bump phpunit/phpunit from 9.6.8 to 9.6.9 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3317
* Bump phpstan/phpstan from 1.10.15 to 1.10.18 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3316
* Fixed display of products count for categories tree by @kyrena in https://github.com/OpenMage/magento-lts/pull/3313
* Some PHPStan fixes for the lib/Varien folder by @fballiano in https://github.com/OpenMage/magento-lts/pull/3271
* Added .gitattributes file by @tmotyl in https://github.com/OpenMage/magento-lts/pull/3318
* Enhanced Google Analytics 4 Integration and Ecommerce Tracking Improvements by @empiricompany in https://github.com/OpenMage/magento-lts/pull/3305
* Added few missing export-ignore to gitattributes by @tmotyl in https://github.com/OpenMage/magento-lts/pull/3320
* Replaced prototypejs load event with vanilla js in js/mage/adminhtml/input-counter.js by @luigifab in https://github.com/OpenMage/magento-lts/pull/3311
* Bump phpstan/phpstan from 1.10.18 to 1.10.19 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3332
* Bump phpseclib/phpseclib from 3.0.19 to 3.0.20 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3333
* Bump friendsofphp/php-cs-fixer from 3.17.0 to 3.18.0 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3334
* Bump macopedia/phpstan-magento1 from 1.0.5 to 1.0.7 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3335
* Fixed passing null for str_replace is deprecated for downloadable products by @kyrena in https://github.com/OpenMage/magento-lts/pull/3330
* Fixed passing null for trim is deprecated in Mage_Catalog_Model_Resource_Category_Flat by @kyrena in https://github.com/OpenMage/magento-lts/pull/3331
* Fixed passing null for unserialize is deprecated in Mage_Sales_Model_Quote_Address by @kyrena in https://github.com/OpenMage/magento-lts/pull/3337
* Fixed "passing null to parameter" in Mage_Core_Helper_String by @kiatng in https://github.com/OpenMage/magento-lts/pull/3341
* getAttributeCode() on null prevention by moving after AttributeAbstract check by @discountscott in https://github.com/OpenMage/magento-lts/pull/3344
* Fixed passing null for trim is deprecated in Mage_Catalog_Model_Attribute_Backend_Customlayoutupdate by @kyrena in https://github.com/OpenMage/magento-lts/pull/3336
* Added methods to support "error state" for cron schedule by @luigifab in https://github.com/OpenMage/magento-lts/pull/3310
* Added README to gitattributes file by @fballiano in https://github.com/OpenMage/magento-lts/pull/3322
* Improvements to Google Analytics 4 code by @empiricompany in https://github.com/OpenMage/magento-lts/pull/3342
* Fixed eav_config_entity throwing error message on removed entity model by @davidhiendl in https://github.com/OpenMage/magento-lts/pull/3338
* Bump tj-actions/changed-files from 36 to 37 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3350
* Bump EnricoMi/publish-unit-test-result-action from 1.40 to 2.7 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3351
* Bump friendsofphp/php-cs-fixer from 3.18.0 to 3.19.2 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3348
* Bump phpstan/phpstan from 1.10.19 to 1.10.21 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3349
* Fixed logException twice in CartController.php by @kiatng in https://github.com/OpenMage/magento-lts/pull/3354
* Removed Google Analytics3 and Universal Analytics by @fballiano in https://github.com/OpenMage/magento-lts/pull/3347
* Fixed getQuote() returns null in quote item and quote address objects. by @kiatng in https://github.com/OpenMage/magento-lts/pull/3355
* Fixed passing null is deprecated for strpos in Mage_Core_Controller_Varien_Action by @kyrena in https://github.com/OpenMage/magento-lts/pull/3327
* Fixed passing null is deprecated for htmlspecialchars in Mage_Core_Helper_Abstract by @kyrena in https://github.com/OpenMage/magento-lts/pull/3324
* Bump friendsofphp/php-cs-fixer from 3.19.2 to 3.20.0 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3360
* Removed support for UPS CGI API which was shut down in May 2023 by @fballiano in https://github.com/OpenMage/magento-lts/pull/3287
* Old inactive quotes are now actually purged from the database by @FredericMartinez in https://github.com/OpenMage/magento-lts/pull/1489
* Added support for Google Tag Manager injection, compatible with Analytics 4 and enhanced ecommerce data by @fballiano in https://github.com/OpenMage/magento-lts/pull/3356
* Added DDEV command to install OM (incl. sample data) by @sreichel in https://github.com/OpenMage/magento-lts/pull/3248
* Allowed customization of the checkout page identifiers for the Mage_GoogleAnalytics module by @fballiano in https://github.com/OpenMage/magento-lts/pull/3363
* Bump phpstan/phpstan from 1.10.21 to 1.10.22 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3359
* Removed unexisting branch alias from composer.json by @fballiano in https://github.com/OpenMage/magento-lts/pull/3364
* Clear config cache after maintenance check for DB Update by @Hanmac in https://github.com/OpenMage/magento-lts/pull/3365
* Added hidden element before multiselect form elements in adminhtml by @Hanmac in https://github.com/OpenMage/magento-lts/pull/3352
* Added decimal/thousand separator specification when sending Analytics4 data by @fballiano in https://github.com/OpenMage/magento-lts/pull/3366
* Added DDEV command to create/update the administrator account by @ADDISON74 in https://github.com/OpenMage/magento-lts/pull/3256
* New feature to repopulate product form in frontend on errors. by @kiatng in https://github.com/OpenMage/magento-lts/pull/3367
* Bump phpstan/phpstan from 1.10.22 to 1.10.25 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3372
* Bump phpseclib/phpseclib from 3.0.20 to 3.0.21 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3369
* Bump friendsofphp/php-cs-fixer from 3.20.0 to 3.21.0 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3370
* Bump magento-ecg/coding-standard from 4.5.2 to 4.5.4 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3371
* Converted line ending of some files in order to avoid warnings from git by @fballiano in https://github.com/OpenMage/magento-lts/pull/3368
* Resized panels height in "attribute set" backend view by @fballiano in https://github.com/OpenMage/magento-lts/pull/3247
* Fixed $address is null in AddressController. by @kiatng in https://github.com/OpenMage/magento-lts/pull/3373
* Fixed exception in Dob.php during customer registration. by @kiatng in https://github.com/OpenMage/magento-lts/pull/3374
* Better clearing of session messages after display by @kyrena in https://github.com/OpenMage/magento-lts/pull/3326
* Do not reindex catalog_product_flat for disabled stores by @empiricompany in https://github.com/OpenMage/magento-lts/pull/3251
* Mage_Report: Migrated hardcoded cron expressions to configurable fields by @Sekiphp in https://github.com/OpenMage/magento-lts/pull/1869
* Mage_Eav_Model_Attribute_Data_Date::extractValue() should not return false by @fballiano in https://github.com/OpenMage/magento-lts/pull/3268
* Fixed customer not receiving order notification email when the email … by @kiatng in https://github.com/OpenMage/magento-lts/pull/3290
* Removed double save on quote in Mage_Checkout_CartController::ajaxUpdateAction() by @fballiano in https://github.com/OpenMage/magento-lts/pull/3378
* Rewrote part of GA4 to support add/remove quantity from cart and cart_updates by @fballiano in https://github.com/OpenMage/magento-lts/pull/3377
* Fixed implicit conversion from float to int in Mage_SalesRule_Model_Validator by @kyrena in https://github.com/OpenMage/magento-lts/pull/3345
* Rewrote afterSave of product and category image attributes by @luigifab in https://github.com/OpenMage/magento-lts/pull/3301
* Bump friendsofphp/php-cs-fixer from 3.21.0 to 3.22.0 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3385
* Bump phpunit/phpunit from 9.6.9 to 9.6.10 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3386
* Fixed passing null for number_format is deprecated for downloadable products by @fballiano in https://github.com/OpenMage/magento-lts/pull/3382
* Allowed to upload images in WebP format by @empiricompany in https://github.com/OpenMage/magento-lts/pull/3384
* Removed phpstan.dist.issues.neon by @sreichel in https://github.com/OpenMage/magento-lts/pull/3257
* Fixed issue #3392 clearInstance() method in Mage_Catalog_Model_Product. by @kiatng in https://github.com/OpenMage/magento-lts/pull/3395
* Bump phpstan/phpstan from 1.10.25 to 1.10.26 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3402
* Fixed a bug with UPS that prevented saving the shipping methods page from backend by @fballiano in https://github.com/OpenMage/magento-lts/pull/3397
* Fixed documentation block typo on Mage::addObserver by @denisahac in https://github.com/OpenMage/magento-lts/pull/3404
* Fixed truecolor detection for WebP images by @fballiano in https://github.com/OpenMage/magento-lts/pull/3403
* Fixed a bug with overlapping menu items in backend by @seifer7 in https://github.com/OpenMage/magento-lts/pull/3396
* Updated copyright docblocks by @fballiano in https://github.com/OpenMage/magento-lts/pull/3411
* Updated PHPStorm meta by @fballiano in https://github.com/OpenMage/magento-lts/pull/3415
* Handled fopen error in Mage_Index_Model_Lock by throwing exception. by @colinmollenhour in https://github.com/OpenMage/magento-lts/pull/3414
* Updating the information about Mage_Poll in the README file by @ADDISON74 in https://github.com/OpenMage/magento-lts/pull/3416

## New Contributors
* @davidhiendl made their first contribution in https://github.com/OpenMage/magento-lts/pull/2993
* @darinda made their first contribution in https://github.com/OpenMage/magento-lts/pull/3065
* @tobihille made their first contribution in https://github.com/OpenMage/magento-lts/pull/2474
* @sidealice made their first contribution in https://github.com/OpenMage/magento-lts/pull/3078
* @Asfolny made their first contribution in https://github.com/OpenMage/magento-lts/pull/1736
* @gorbunovav made their first contribution in https://github.com/OpenMage/magento-lts/pull/3276
* @Tomasz-Silpion made their first contribution in https://github.com/OpenMage/magento-lts/pull/3274
* @speedupmate made their first contribution in https://github.com/OpenMage/magento-lts/pull/1210
* @Judx made their first contribution in https://github.com/OpenMage/magento-lts/pull/2673
* @kyrena made their first contribution in https://github.com/OpenMage/magento-lts/pull/3299
* @pquerner made their first contribution in https://github.com/OpenMage/magento-lts/pull/3307
* @Hanmac made their first contribution in https://github.com/OpenMage/magento-lts/pull/3365
* @denisahac made their first contribution in https://github.com/OpenMage/magento-lts/pull/3404
* @seifer7 made their first contribution in https://github.com/OpenMage/magento-lts/pull/3396

**Full Changelog**: https://github.com/OpenMage/magento-lts/compare/v20.0.20...v20.1.0

<hr /><em>This discussion was created from the release <a href='https://github.com/OpenMage/magento-lts/releases/tag/v20.1.0'>v20.1.0</a>.</em>

**Comment by colinmollenhour** [comment]
Awesome work, everyone! Special thanks for putting the release notes together @fballiano !

**Reply by addison74** [reply]
We should mention that he became the contributor with the most commits ;)

**Reply by fballiano** [reply]
I do many maintenance commits like version bumps so... you know :-D
Thanks for noticing, but if we manage to do something good it's because we're a group, small, but still a group heheheh

**Reply by sreichel** [reply]
Cheers ...

**Comment by Hanmac** [comment]


**Reply by fballiano** [reply]
we have to push people to upgrade to v20, we can't be tied to explorer 8 in 2023, we need to move forward and let everybody understand that openmage is not only babysitting a dieing magento 1. this is good in the long run.

I also hope we'll release v21 soon and we start really distancing ourselves from the past. I'm tired of hearing "we will migrate to another platform next year".

**Reply by sreichel** [reply]
"we have to push" ... HELL NO!

You never cared about shops that just wanted to migrate to OM.  Dropped support for v19 branch was such a big mistake imho ... but hey, its you personal playground now. Was a good descision to step back ...

Seems you are only acting on developers view (that i agree with for "next"), but not on store owners with limited resources ...

**Reply by justinbeaty** [reply]
There was an RFC, and the majority of us decided to drop v19. It’s unfair to say @fballiano is treating this as anything close to his personal playground. 🤷‍♂️

**Reply by sreichel** [reply]
@justinbeaty OM was ment to be a drop-in replacemant for M1 ... the majority is/was part for what time?

**Reply by justinbeaty** [reply]
I’m sorry, but why is it okay to denigrate fabrizio for following the RFC, no matter what the original intention of this project was?

**Reply by sreichel** [reply]
@justinbeaty "I'm just sad that my commits aren't backported into 19.x" ... same for some of my PRs.

It feels wong. They project does not follow its intention anymore. 

No matter = fork it.

**Reply by justinbeaty** [reply]
You did not answer the question as to why it is okay to attack @fballiano for carrying out an RFC (that he didn’t even create) that passed with an overwhelming majority vote.

Why not start an RFC proposing to revert things how you’d like them, or fork the project yourself?

**Reply by sreichel** [reply]
Think about how long I contributed to the project? For its orignal intention.

Are you telling me that the quality has improved in recent months?

Edit: "why it is okay" ... b/c this project is not community-driven anymore! It realy feels like a personal playground now ...

**Reply by justinbeaty** [reply]
This will be my last reply, it’s late here.

I’ve been around this project since 2017, and back then drop in replacement was a big goal. But I can guarantee you I can find references from back then that at some point the project would evolve past solely bug fixes. So tbh, there really was no strict original intention.

Yes, you have contributed a lot. But you’re not the only person here.

Yes, I am happy with the direction of the project. There are a lot of forward thinking enhancements that will keep OM alive and relevant.

It’s still not okay to attack someone for what was **in fact a community decision,** no matter how much you disagree with it.

**Reply by Flyingmana** [reply]
> Think about how long I contributed to the project? For its orignal intention.
> 
> Are you telling me that the quality has improved in recent months?
> 
> Edit: "why it is okay" ... b/c this project is not community-driven anymore! It realy feels like a personal playground now ... 

The RFC was a community decision.
The community can change this with a new RFC.

I have to give you an official warning for your behavior here.
This is no place for complaining about made decisions where the community was involved with a long timeframe for comments.

If you are not satisfied, you can follow the process, or be the change outside of the Org via a Fork.
If you continue to complain on unrelated places inside the OpenMage org, I will from now on make use of temporary Bans.

**Reply by Hanmac** [reply]
this isn't about backporting new features, this is about backporting critical bugs that can prevent Database updates

**Reply by fballiano** [reply]
@Hanmac I may be mistaken but I thought the RFC was talking about backporting only security patches. Anyway, I did what I though was compliant with the rule and in some cases I've asked the other members if something should be backported or not since I couldn't have an opinion. But anyway there are many maintainers, it's not only me, anybody in the group can do that. I'm already dedicating all of my time to openmage, but I'm freaking not omnipotent eh, if I could ever get a bit of help instead of critics for dedicating at least 5 hours per day to try to do something with this project!

**Reply by Flyingmana** [reply]
> @Hanmac I may be mistaken but I thought the RFC was talking about backporting only security patches. Anyway, I did what I though was compliant with the rule and in some cases I've asked the other members if something should be backported or not since I couldn't have an opinion. But anyway there are many maintainers, it's not only me, anybody in the group can do that. I'm already dedicating all of my time to openmage, but I'm freaking not omnipotent eh, if I could ever get a bit of help instead of critics for dedicating at least 5 hours per day to try to do something with this project!
> 
 @Hanmac 
to rephrase this a bit, at the moment we dont have the resources for proactive backporting.
But if community members do the backporting via PRs, we are going to accept them uncomplicated and will do a bugfix release there too.

**Reply by Flyingmana** [reply]
you got the warning, now you need to accept the 7day block

**Comment by sreichel** [comment]
"Uh oh! You can't vote right now."

@fballiano seriously? I cant even vote here anymore? You have a way to much permissions.

**Reply by Flyingmana** [reply]
For protocol, this is not prevented because of any Org setting.
and if its because of personal interaction limits, we dont have any rule which requires maintainers to be open for any interaction.

**Reply by fballiano** [reply]
@sreichel since a few days ago I blocked you **on my personal account** because of the way you attack me since months without doing anything constructively. This has cause extreme stress to me and that's the simple reason why you can't do some interactions with what I post, again a **personal setting**, nothing to do with the openmage repository.
For the last time I'll say that I defended you and trusted you against other maintainers and you treated me like s**t anyway so, personally, you're lost my trust and I'm done with you.

**Reply by sreichel** [reply]
Again ... i've no time to contribute, but i still take some time for reviews! I'll not approve anything without tests, ... so I can only  "criticize" atm. Detailed anwser at @ADDISON74 post ... later ...

**Comment by sreichel** [comment]
"I also hope we'll release v21 soon and we start really distancing ourselves from the past."

Fork this project for your ideas!

**Reply by fballiano** [reply]
Can't I even have a personal hope for this project without being told to go away?

I've always being very open about my visions for openmage, but I've always acted conforming the RFCs, and when I wanted to change the rules I wrote and promoted RFCs as I was told by other maintainers.
I'm more then willing to change my contributions here or to fix my mistakes if I'm told so by the rest of the maintainers.

**Reply by sreichel** [reply]
"I've always being very open about my visions for openmage ..."

Please allow me the time for a serious answer. Okay?

**Comment by addison74** [comment]
The post is about the release of a new version. I propose to have the discussions related to RFC, v19, in other post here https://github.com/OpenMage/magento-lts/discussions/3424.

I ask the owners to delete all the off-topic discussions.
